### PR TITLE
Fixing coordinate system syntax

### LINKED ISSUE #6765: When users specify blocks and no coordinate system, MOOSE would not let them run their simulation
### Description of the enhancement or error report

When users specify blocks and no coordinate system, MOOSE stops with an error.

### Rationale for the enhancement or information for reproducing the error

This setup should be interpreted as "use the default coordinate system on all of my blocks" and not as an error.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

Will make MOOSE smarter!

### PR DESCRIPTION
**Pull request by andrsd**
The "coord_type" parameter was `MultiMooseEnum` which was incorrect, because we have to be able to specify one value multiple times. One entry per block ID.

Another issue related to this problem that this MR is fixing is that when users specify `block` in `[GlobalParams]`, MOOSE sees mismatch between blocks and coord_sys. Thus an additional logic was included to prevent this. When people specify multi blocks and just one coord_type, we will set all these blocks to this coord_type.

Closes #6765

### CODE REVIEW

**Comment by dschwen** [comment]
I thought it preserved duplicate values, and it does have a `std::vector<std::string>` member for that, but there is no _sane_ way of fetching those. IMO `MultiMooseEnum::operator[]()` is _hosed_ (the assert in particular) and needs some fixing!

**Comment by andrsd** [comment]
> I thought it preserved duplicate values

I think you are right.

**Comment by andrsd** [comment]
Let me try to use the `MultiMooseEnum` way...

**Comment by andrsd** [comment]
This version keeps the `MultiMooseEnum` and is just fixing the logic in `FEProblem`.

File: test/tests/coord_type/tests [hunk 1]
```diff
@@ -5,7 +5,7 @@
     input = coord_type_rz.i
     exodiff = coord_type_rz_out.e
   [../]
-  [/rz-x-rotation]
+  [./rz-x-rotation]
```
Comment (permcody) [inline test/tests/coord_type/tests]: Looks like we need some error checking in the TestHarness. This would have triggered an error in MOOSE.

**Pull request merged**